### PR TITLE
Enhancement-share-simulation

### PR DESCRIPTION
**Pull request by joachimhgg**
**Description**
 Add a checkbox option to save the scene or the animation locally. The user thus needs to choose if they wants the simulation be uploaded or saved on local and cannot do both from Webots.

### CODE REVIEW

File: src/webots/gui/WbMainWindow.cpp [hunk 1]
```diff
@@ -1574,64 +1579,45 @@ void WbMainWindow::exportVrml() {
...
-  WbShareWindow *shareWindowUi = new WbShareWindow(this);
-  shareWindowUi->exec();
+  WbShareWindow *shareWindow = new WbShareWindow(this);
+  shareWindow->exec();
   WbSimulationState::instance()->setMode(currentMode);
 }
 
 void WbMainWindow::uploadScene() {
+  QString filename;
+  filename = exportHtmlFiles();
```
Comment (omichel) [inline src/webots/gui/WbMainWindow.cpp:1601]:
```suggestion
  const QString filename = exportHtmlFiles();
```

File: src/webots/gui/WbMainWindow.cpp [hunk 2]
```diff
@@ -2376,10 +2362,14 @@ void WbMainWindow::setWorldLoadingStatus(const QString &status) {
 }
 
 void WbMainWindow::startAnimationRecording() {
+  QString filename = exportHtmlFiles();
```
Comment (omichel) [inline src/webots/gui/WbMainWindow.cpp:2365]:
```suggestion
  const QString filename = exportHtmlFiles();
```

File: src/webots/gui/WbShareWindow.cpp [hunk 3]
```diff
@@ -58,6 +59,14 @@ WbShareWindow::WbShareWindow(QWidget *parent) : QDialog(parent) {
   layout->addWidget(pushButtonScene, 5, 0, 1, 1);
 
   WbMainWindow *mainWindow = dynamic_cast<WbMainWindow *>(parentWidget());
+
+  QCheckBox *checkBoxSave = new QCheckBox(this);
+  checkBoxSave->setFocusPolicy(Qt::NoFocus);
+  checkBoxSave->setText(tr("Save only on local"));
```
Comment (omichel) [inline src/webots/gui/WbShareWindow.cpp:65]:
```suggestion
  checkBoxSave->setText(tr("Save as local files"));
```

File: src/webots/gui/WbMainWindow.cpp [hunk 4]
```diff
@@ -1737,9 +1723,9 @@ void WbMainWindow::uploadFinished() {
   } else {
     WbLog::info(tr("link: %1\n").arg(url));
 
-    mLinkWindow = new WbLinkWindow(this);
-    mLinkWindow->setLabelLink(url);
-    mLinkWindow->exec();
+    WbLinkWindow *linkWindow = new WbLinkWindow(this);
```
Comment (omichel) [inline src/webots/gui/WbMainWindow.cpp:1726]: I am afraid there is a memory leak here... It seems this object should get deleted when we delete the WbMainWindow object, but we should probably delete it earlier when this linkWindow is not needed any more. Is it the case?

File: src/webots/gui/WbMainWindow.cpp [hunk 5]
```diff
@@ -196,7 +196,12 @@ WbMainWindow::WbMainWindow(bool minimizedOnStart, WbStreamingServer *streamingSe
   WbAnimationRecorder *recorder = WbAnimationRecorder::instance();
   connect(recorder, &WbAnimationRecorder::initalizedFromStreamingServer, this, &WbMainWindow::disableAnimationAction);
   connect(recorder, &WbAnimationRecorder::cleanedUpFromStreamingServer, this, &WbMainWindow::enableAnimationAction);
-  connect(recorder, &WbAnimationRecorder::requestOpenUrl, this, [this]() { this->upload('A'); });
+  connect(recorder, &WbAnimationRecorder::requestOpenUrl, this, [this](QString filename, QString content, QString title) {
```
Comment (omichel) [inline src/webots/gui/WbMainWindow.cpp:199]:
```suggestion
  connect(recorder, &WbAnimationRecorder::requestOpenUrl, this, [this](const QString &filename, const QString &content, const QString &title) {
```

File: src/webots/gui/WbMainWindow.cpp [hunk 6]
```diff
@@ -1574,64 +1579,45 @@ void WbMainWindow::exportVrml() {
...
 }
 
-void WbMainWindow::exportHtmlFiles() {
-  WbSimulationState::Mode currentMode = WbSimulationState::instance()->mode();
-
-  QString fileName = findHtmlFileName("Export HTML File");
-  if (fileName.isEmpty()) {
+QString WbMainWindow::exportHtmlFiles() {
+  QString filename = WbStandardPaths::webotsTmpPath() + "cloud_export.html";
+  if (mSaveCheckboxStatus) {
```
Comment (omichel) [inline src/webots/gui/WbMainWindow.cpp:1584]:
```suggestion
  QString filename;
  if (!mSaveCheckboxStatus)
    filename = WbStandardPaths::webotsTmpPath() + "cloud_export.html";
  else {
```

**Review by omichel — CHANGES_REQUESTED** [review]
It seems pretty good, except for coding style (const-correctness and double initialization).

File: src/webots/gui/WbMainWindow.cpp (hunk 4 above)
Comment (joachimhgg) [reply src/webots/gui/WbMainWindow.cpp:1727]: It should be fixed by not creating it using the `new` syntax

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**